webservices/resources/legal.py

```diff
@@ -4,19 +4,31 @@
 
 es = utils.get_elasticsearch_connection()
 
+class AdvisoryOpinion(utils.Resource):
+    # @use_kwargs(args.ao_no)
+    def get(self, ao_no):
+        query = {"query": {"bool": {"must": [{"term": {"no": ao_no}},
+                          {"term": {"_type": "advisory_opinions"}}]}},
+                          "_source": {"exclude": "text"}}
+
+        es_results = es.search(query)
+
+        results = {"docs": [r["_source"] for r in es_results["hits"]["hits"]]}
+        return results
+
 class Search(utils.Resource):
     @use_kwargs(args.query)
-    def get(self, q, from_hit=0, hits_returned=20, type='all', **kwargs):
-        if type == 'all':
+    def get(self, q=None, from_hit=0, hits_returned=20, _type='all', **kwargs):
```

Comment:
Did you mean to include `q=None` here? I think eventually that's right, but we didn't change the rest of the code to handle `None` differently.